src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1316,11 +1498,24 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::GetWithParam(spinel_prop
 
     assert(mWaitingTid == 0);
 
-    mPropertyFormat = aFormat;
     va_start(mPropertyArgs, aFormat);
-    error = Request(true, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam, aParamSize);
+    error =
+        RequestWithPropertyFormat(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam, aParamSize);
     va_end(mPropertyArgs);
-    mPropertyFormat = nullptr;
+
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_ENABLE
+    if (mRcpFailed)
+    {
+        RecoverFromRcpFailure();
+
+        va_start(mPropertyArgs, aFormat);
+        error = RequestWithPropertyFormat(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, SPINEL_DATATYPE_DATA_S, aParam,
+                                          aParamSize);
+        va_end(mPropertyArgs);
+
+        VerifyOrDie(!mRcpFailed, OT_EXIT_FAILURE);
```

Comment:
With current model, seems like we allow one RCP failure (if it happens again while recovering) we fail hard and `DieNow`). 

- Can we relax this to allow `x` number of back-to-back failure before stop (say 3 or 4), instead of 2.
- I see the same pattern of code (i.e. check if RCP failed, recover and the try the same command again) in all different methods (`Get()`, `Set()`, `Insert()`, ...). It would be good can we somehow re-use this pattern/code (we are calling `Insert()` and if it fails, we want to recover and then try `Insert()` again with same params). I don't have a clear suggestion on how we may do it (it's fine for me as is), just raising it so we can think about it.

Reply:
Agree.

Reply:
Agreed with retrying more than once.

> Another way to set the retry time might be changing the configuration

This looks good to me.

For the reusing of code, I did some refactoring <https://github.com/openthread/openthread/pull/5694#discussion_r518504037>. This seems the best I can do.

Reply:
Updated. Now using `OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT` to control max failure count allowed.